app.py

```diff
@@ -48,52 +123,96 @@ def process_upload_function(event, context):
+    # Get the uploaded file's information
+    messages = batch_retrieval_from_sqs()
+    if not messages:
         return
 
-    # Get the bytes from S3
-    datadir = "/tmp/"
+    bucket = messages[0].get("bucket")
 
-    key_list = key.split("/")
+    for message in messages:
+        bucket = message.get("bucket")
+        key = message.get("file_path")
+        jurisdiction_id = message.get("jurisdiction_id")
 
-    jurisdiction_acronym = key_list[0]  # e.g az, al, etc
+        # for some reason, the key is url encoded sometimes
+        key = urllib.parse.unquote(key, encoding="utf-8")
 
-    name_of_file = key_list[-1]  # e.g. bills.json, votes.json, etc
-    filedir = f"{datadir}{name_of_file}"
+        # we want to ignore the trigger for files that are dumped in archive
+        if key.startswith("archive"):
+            return
 
-    s3_client.download_file(
-        bucket, key, filedir
-    )  # Download this file to writable tmp space.
+        # Get the bytes from S3
 
-    jurisdiction_id = (
-        f"ocd-jurisdiction/country:us/state:{jurisdiction_acronym}/government"
-    )
+        key_list = key.split("/")
+
+        jurisdiction_abbreviation = key_list[0]  # e.g az, al, etc
+
+        # we want to filter out unique jurisdiction
+
+        if jurisdiction_abbreviation not in unique_jurisdictions:
+            unique_jurisdictions[jurisdiction_abbreviation] = {
+                "id": jurisdiction_id,
+                "keys": [],
+            }
+        unique_jurisdictions[jurisdiction_abbreviation]["keys"].append(key)
+
+        # name_of_file = key_list[-1]  # e.g. bills.json, votes.json, etc
+
+        filedir = f"{datadir}{key}"
+
+        # Check if the directory for jurisdiction exists
+        dir_path = os.path.join(datadir, jurisdiction_abbreviation)
+        if not os.path.exists(dir_path):
+            os.makedirs(dir_path)
+
+        all_files.add(filedir)
+        # Download this file to writable tmp space.
+        try:
+            s3_client.download_file(bucket, key, filedir)
+        except Exception as e:
+            logger.error(f"Error downloading file: {e}")
+            all_files.remove(filedir)
+            continue
+
+    # Process imports for all files per jurisdiction in a batch
+    for abbreviation, juris in unique_jurisdictions.items():
 
-    logger.info(f"importing {jurisdiction_id}...")
-    do_import(jurisdiction_id, datadir)
+        logger.info(f"importing {juris['id']}...")
+        try:
+            do_import(juris["id"], f"{datadir}{abbreviation}")
+        except Exception as e:
+            logger.error(f"Error importing {juris['id']}: {e}")
+            continue
 
-    try:
-        os.remove(filedir)
-    except Exception as e:
+        archive_files(bucket, juris["keys"])
 
-        logger.warning(f"Failed to remove file :: {e}")
-    finally:
-        logger.info(">>>> DONE IMPORTING <<<<")
+    for file in all_files:
+        try:
+            os.remove(file)
```

Comment:
Good habit to clean up the files, but if they're in a temp space within the lambda, do we _need_ to clean them up, or do they get cleaned up by the lambda being stopped?

Reply:
We don't need to delete them since it's tmp space